packages/api/src/models/animalBatchModel.js

```diff
@@ -51,9 +51,21 @@ class AnimalBatchModel extends baseModel {
         id: { type: 'integer' },
         name: { type: 'string' },
         notes: { type: ['string', 'null'] },
+        removed: { type: 'boolean' },
```

Comment:
I know its in the ticket but is this necessary? Could we not just:

`animal_removal_reason_id = null === removed= false`
and
`animal_removal_reason_id != null === removed= true`

This has the added benefit of not needing constraints or checks

Reply:
Oh my gosh I think I love that idea 😂  Let me run through the logic in the PATCH controllers + tests to see if holds up cleanly throughout the whole flow.

Reply:
@antsgar what do think? It's super easy to implement on the backend (particularly if we don't care about keeping `animal_removal_reason_id` and the optional string `removal_explanation` in sync, which the current code doesn't do anyway.

The only drawback I can think of is that throughout the app we're going to be checking/filtering on `animal.animal_removal_reason_id` which is awkward compared to `animal.removed`. We could generate a `removed` property on the frontend, but I feel like making the RTK Query-managed entity start to diverge from the backend like that would be a mistake that would just confuse us later... if we want a `removed` property I think it should be in the database, and @Duncan-Brain is right, we would need the opposite check on it too.

So I could go either way with this...!